### ISSUE TITLE
[bug] default_options are typed to unicode

### ISSUE THREAD

**Issue by ofiriluz** [conan-io/conan#6321]
### Environment Details (include every applicable attribute)
  * Operating System+version: solaris intel or solaris ultrasparc (any version)
  * Compiler+version: gcc 4.8.5
  * Conan version: 1.21.0
  * Python version: 2.7.16 (Deprecation, but an issue to upgrade in our environment)

### Steps to reproduce (Include if Applicable)
when setting the following default options as:
```python
default_options=("SOME_OPT=%s" % (os.environ["SOME_OPT"]))
```
It causes the value reaching the conan_file.py create_options to be unicode, causing an exception
You can fix this by changing line 26 on model/conan_file.py from
```python
elif isinstance(default_options, str)
```
to be
```python
elif isinstance(default_options, (str, unicode))
```

Would be much appriciated and a quick fix :)

Thanks,
Ofir

**Comment by jgsogo** [comment]
I'm not able to reproduce this issue in the CI, but I think this can be changed without any risk for the next minor release, even though we are about to deprecate python2. I've opened a PR with the changes for the next release `v1.21.1`.

---

Anyway, I would really like to know if it is a real issue to upgrade your environment to Python 3, or you are just delaying that change as much as possible. We really think that we should deprecate Conan for py2 too, so if there is a real stopper we would love to know it asap. Thanks!

**Comment by ofiriluz** [comment]
Hey,

We are in the process of upgrading our build servers, this is a task in hand but we have other priorities and need the build servers to keep working since we are supporting many different architectures such as HP-UX AIX and so on.

we will get there eventually :)

Thank you for the quick reply!

**Comment by jgsogo** [comment]
Great to know that you will eventually be using py3 and it is not a blocker 👍   There are so many environments out there for C++ and we are afraid that some of them might not be able to move to Python3

**Comment by memsharded** [comment]
Merged fix, will be released next week in Conan 1.21.1
